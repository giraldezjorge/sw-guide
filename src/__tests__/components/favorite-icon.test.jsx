import React from 'react'
import renderer from 'react-test-renderer'
import { render, mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import FavoriteIcon from '../../components/favorite-icon'

const mockStore = configureStore([])

describe('The favorite icon', () => {
  const item = {
    url: 'test_url'
  }

  it('shows the add favorite icon if item is not a favorite', () => {
    const store = mockStore({})
    const component = renderer.create(
      <Provider store={store}>
        <FavoriteIcon favorite={false} item={item} />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it('shows the remove favorite icon if item is a favorite', () => {
    const store = mockStore({
      favorites: [item]
    })
    const component = renderer.create(
      <Provider store={store}>
        <FavoriteIcon favorite={true} item={item} />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
