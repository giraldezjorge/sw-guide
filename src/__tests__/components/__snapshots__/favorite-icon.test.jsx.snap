// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The favorite icon shows the add favorite icon if item is not a favorite 1`] = `
<a
  className="button is-small is-danger is-outlined"
  onClick={[Function]}
  role="button"
  title="Add to favorites"
>
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-heart"
    />
  </span>
</a>
`;

exports[`The favorite icon shows the remove favorite icon if item is a favorite 1`] = `
<a
  className="button is-small is-danger "
  onClick={[Function]}
  role="button"
  title="Remove from favorites"
>
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-heart"
    />
  </span>
</a>
`;
